### ISSUE TITLE
Stack Fails to build due to The following resource(s) failed to create: [AnsibleConfigServer]

### ISSUE THREAD

**Issue by mandibuswell** [aws-quickstart/quickstart-redhat-openshift#135]
After trying the new build that kicks off the ansible server earlier in the process, the stack is failing to build.
The ansible config server is successfully started earlier and in fact builds. 

AWS::EC2::Instance | AnsibleConfigServer | Resource creation Initiated
-- | -- | --

It is reachable. however at the very end of the process after the OpenShift Masters are created, there is a final step trying to create the Ansible Config Server again, at this point it appears to time out after a long period of time

17:21:51 UTC+1200 | CREATE_FAILED | AWS::CloudFormation::Stack | Red-Hat-OpenShift-NewVersion-OpenShiftStack-1KSG4QESM4WA0 | The following resource(s) failed to create: [AnsibleConfigServer].
-- | -- | -- | -- | --
  | 17:21:49 UTC+1200 | CREATE_FAILED | AWS::EC2::Instance | AnsibleConfigServer | Received FAILURE signal with UniqueId i-0097899e0d3e8a3c4
  | 16:42:37 UTC+1200 | CREATE_COMPLETE | AWS::AutoScaling::AutoScalingGroup | OpenShiftMasterASG

I can confirm that the cluster is up and functioning. I think its trying to launch the config server again at the end and that is causing it to fail

**Comment by GeorgeB66** [comment]
I think this might be related, ansible-config server appears to hang in the cloudformation "creating" stage . By ssh'ing onto the ansible server we saw this in the logs:

> An error occurred (AccessDenied) when calling the DescribeStackEvents operation: User: arn:aws:sts::REDACTEDFORGITHUB:assumed-role/GB-OCP-OpenShiftStack-1UBB211CYKT6I-SetupRole-20D3GFLR2ETH/i-0e9b5d1daae0b7eeb is not authorized to perform: cloudformation:DescribeStackEvents on resource: arn:aws:cloudformation:eu-central-1:REDACTEDFORGITHUB:stack/GB-OCP-OpenShiftStack-1UBB211CYKT6I/*

We checked the IAM permissions assigned to the ansible-config server and all appeared ok, however the error persisted and running the aws cli from the ansible server to DescribeStackEvents kept failing. In the end I reduced the granularity of the IAM permissions so that the ID of the stack was set to any, and immediately it started working.

Unfortunately 5 minutes later when the deploy playbook was called it then failed for another reason (still to diagnose).

**Comment by mikecali** [comment]
@GeorgeB66, did you managed to check if your stack created 2 Ansible-Config servers just like what I have seen?

**Comment by mirwin77** [comment]
I ran in to the same issue as @GeorgeB66 and was able to successfully complete the stack after manually elevating the IAM permissions for the SetupRole.

**Comment by jaymccon** [comment]
I've been entirely unable to replicate this, either manually or in CI. 

If possible could someone who has experienced it provide me with the region launched in, a set of (redacted) parameters used at launch, as well as logs from the ansible server (please provide a few more lines around any error messages for context as to what command generated the error).

**Comment by emcfins** [comment]
I am having the same issue with the Ansible instance: 
![screen shot 2018-06-26 at 11 31 38 am](https://user-images.githubusercontent.com/11314145/41923053-81cabf48-7934-11e8-955b-6092d11b8c95.png)

I sshed into the instance and checked the cloud-init.log, it looks like it's completing successfully:
```2018-06-14 19:47:02,567 - handlers.py[DEBUG]: finish: modules-final/config-power-state-change: SUCCESS: config-power-state-change ran successfully
2018-06-14 19:47:02,567 - main.py[DEBUG]: Ran 10 modules with 0 failures
2018-06-14 19:47:02,568 - util.py[DEBUG]: Creating symbolic link from '/run/cloud-init/result.json' => '../../var/lib/cloud/data/result.json'
2018-06-14 19:47:02,568 - util.py[DEBUG]: Reading from /proc/uptime (quiet=False)
2018-06-14 19:47:02,568 - util.py[DEBUG]: Read 15 bytes from /proc/uptime
2018-06-14 19:47:02,568 - util.py[DEBUG]: cloud-init mode 'modules' took 528.888 seconds (528.89)
2018-06-14 19:47:02,568 - handlers.py[DEBUG]: finish: modules-final: SUCCESS: running modules for final
```
@jaymccon - I can provide you access to this instance if that would help.

**Comment by mirwin77** [comment]
I ran into this issue again today.  See the logs below.  I was able to fix by adding AWSCloudFormationReadOnlyAccess to the SetupRole in IAM permissions.  Let me know if you need any more info.

`Jun 28 10:02:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: ++ grep -c OpenShift
Jun 28 10:02:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: ++ aws cloudformation describe-stack-events --stack-name ITS-RedHat-OpenShift --region us-east-2 --query 'StackEvents[?ResourceStatus == `CREATE_COMPLETE` && ResourceType == `AWS::AutoScaling::AutoScalingGroup`].LogicalResourceId' --output json
Jun 28 10:02:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: An error occurred (AccessDenied) when calling the DescribeStackEvents operation: User: arn:aws:sts::729272666017:assumed-role/ITS-RedHat-OpenShift-SetupRole-IW9D3ZC7125J/i-001c05df37b7f2a4d is not authorized to perform: cloudformation:DescribeStackEvents on resource: arn:aws:cloudformation:us-east-2:729272666017:stack/ITS-RedHat-OpenShift/*
Jun 28 10:02:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: + '[' 0 -lt 3 ']'
Jun 28 10:02:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: + echo 'Waiting for ASG'\''s to complete provisioning...'
Jun 28 10:02:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: Waiting for ASG's to complete provisioning...
Jun 28 10:02:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: + sleep 120
Jun 28 10:03:25 ip-10-2-26-168.us-east-2.compute.internal run-parts(/etc/cron.hourly)[23712]: finished 0yum-hourly.cron
Jun 28 10:04:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: ++ aws cloudformation describe-stack-events --stack-name ITS-RedHat-OpenShift --region us-east-2 --query 'StackEvents[?ResourceStatus == `CREATE_COMPLETE` && ResourceType == `AWS::AutoScaling::AutoScalingGroup`].LogicalResourceId' --output json
Jun 28 10:04:21 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: ++ grep -c OpenShift
Jun 28 10:04:22 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: An error occurred (AccessDenied) when calling the DescribeStackEvents operation: User: arn:aws:sts::729272666017:assumed-role/ITS-RedHat-OpenShift-SetupRole-IW9D3ZC7125J/i-001c05df37b7f2a4d is not authorized to perform: cloudformation:DescribeStackEvents on resource: arn:aws:cloudformation:us-east-2:729272666017:stack/ITS-RedHat-OpenShift/*
Jun 28 10:04:22 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: + '[' 0 -lt 3 ']'
Jun 28 10:04:22 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: + echo 'Waiting for ASG'\''s to complete provisioning...'
Jun 28 10:04:22 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: Waiting for ASG's to complete provisioning...
Jun 28 10:04:22 ip-10-2-26-168.us-east-2.compute.internal cloud-init[1189]: + sleep 120`

**Comment by jsm174** [comment]
@jaymccon I am now seeing this issue too. 

Prior to June 28, I was able to create and teardown several stacks as I worked on scripting:

```
aws cloudformation create-stack \
   --capabilities=CAPABILITY_IAM \
   --stack-name ${OS_PROJECT}-openshift \
   --template-url https://aws-quickstart.s3.amazonaws.com/quickstart-redhat-openshift/templates/openshift-master.template \
   --parameters "ParameterKey=AvailabilityZones,ParameterValue='${AWS_OS_AVAILABILITY_ZONES}'" \
                ParameterKey=ContainerAccessCIDR,ParameterValue="0.0.0.0/0" \
                ParameterKey=RemoteAccessCIDR,ParameterValue="0.0.0.0/0" \
                ParameterKey=KeyPairName,ParameterValue="${OS_PROJECT}-key" \
                ParameterKey=DomainName,ParameterValue="${AWS_ROUTE53_DOMAIN}" \
                ParameterKey=HostedZoneID,ParameterValue="${AWS_ROUTE53_HOSTED_ZONE_ID}" \
                ParameterKey=SubDomainPrefix,ParameterValue="${OS_PROJECT}" \
                ParameterKey=RedhatSubscriptionPoolID,ParameterValue="${REDHAT_POOL_ID}" \
                ParameterKey=RedhatSubscriptionUserName,ParameterValue="${REDHAT_USERNAME}" \
                ParameterKey=RedhatSubscriptionPassword,ParameterValue="${REDHAT_PASSWORD}" \
                ParameterKey=OpenShiftAdminPassword,ParameterValue="${OS_ADMIN_PASSWORD}" \
                ParameterKey=HawkularMetrics,ParameterValue="Disabled"  
```

Last night, I took down the stack, and removed the systems from the RH portal.  I then went to create a new stack and started receiving the error.

After adding `--disable-rollback` and SSH'ing into the `AnsibleConfigServer` 

```
/var/log/messages:

Jul 11 13:11:32 ip-10-0-128-212 ansible-file: Invoked with directory_mode=None force=False remote_src=None _original_basename=None path=/tmp/etcd_certificates-jvVJKhL owner=None follow=True group=None unsafe_writes=None serole=None content=NOT_LOGGING_PARAMETER state=absent setype=None selevel=None regexp=None src=None seuser=None recurse=False _diff_peek=None delimiter=None mode=None attributes=None backup=None
Jul 11 13:13:08 ip-10-0-128-212 cloud-init: ERROR! |failed expects hostvars is a dict
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: + qs_err ' Openshift installation failed, see logs for details '
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: + touch /var/tmp/stack_failed
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: + echo '[FAILED] @  Openshift installation failed, see logs for details '
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: + echo '[FAILED] @  Openshift installation failed, see logs for details '
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: [FAILED] @  Openshift installation failed, see logs for details
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: ++ qs_status
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: ++ '[' -f /var/tmp/stack_failed ']'
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: ++ printf 1
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: ++ return 1
Jul 11 13:13:09 ip-10-0-128-212 cloud-init: + cfn-signal -e 1 --stack myuscis1-openshift-OpenShiftStack-1588DK7OO64YH --resource AnsibleConfigServer --region us-east-1
Jul 11 13:13:09 ip-10-0-128-212 ec2:

/var/log/ansible.log:

2018-07-11 13:10:05,635 p=23348 u=root |  CHECK [memory_availability : ip-10-0-51-180.ec2.internal] **********************
2018-07-11 13:10:06,571 p=23348 u=root |  CHECK [package_version : ip-10-0-92-227.ec2.internal] **************************
2018-07-11 13:10:06,571 p=23348 u=root |  CHECK [docker_image_availability : ip-10-0-92-227.ec2.internal] ****************
2018-07-11 13:10:06,572 p=23348 u=root |  CHECK [memory_availability : ip-10-0-92-227.ec2.internal] **********************
2018-07-11 13:10:08,998 p=23348 u=root |  CHECK [disk_availability : ip-10-0-2-248.ec2.internal] *************************
2018-07-11 13:10:08,998 p=23348 u=root |  CHECK [package_availability : ip-10-0-2-248.ec2.internal] **********************
2018-07-11 13:10:11,080 p=23348 u=root |  CHECK [package_version : ip-10-0-2-248.ec2.internal] ***************************
2018-07-11 13:10:12,167 p=23348 u=root |  CHECK [docker_image_availability : ip-10-0-2-248.ec2.internal] *****************
2018-07-11 13:10:39,963 p=23348 u=root |  CHECK [memory_availability : ip-10-0-2-248.ec2.internal] ***********************
2018-07-11 13:13:08,935 p=23348 u=root |  ERROR! |failed expects hostvars is a dict

/var/log/cloud-init.log:

2018-07-11 13:13:09,498 - handlers.py[DEBUG]: start: modules-final/config-power-state-change: running config-power-state-change with frequency once-per-instance
2018-07-11 13:13:09,498 - util.py[DEBUG]: Writing to /var/lib/cloud/instances/i-0bba4695fbe893afe/sem/config_power_state_change - wb: [420] 19 bytes
2018-07-11 13:13:09,499 - util.py[DEBUG]: Restoring selinux mode for /var/lib/cloud/instances/i-0bba4695fbe893afe/sem/config_power_state_change (recursive=False)
2018-07-11 13:13:09,499 - util.py[DEBUG]: Restoring selinux mode for /var/lib/cloud/instances/i-0bba4695fbe893afe/sem/config_power_state_change (recursive=False)
2018-07-11 13:13:09,499 - helpers.py[DEBUG]: Running config-power-state-change using lock (<FileLock using file '/var/lib/cloud/instances/i-0bba4695fbe893afe/sem/config_power_state_change'>)
2018-07-11 13:13:09,499 - cc_power_state_change.py[DEBUG]: no power_state provided. doing nothing
2018-07-11 13:13:09,499 - handlers.py[DEBUG]: finish: modules-final/config-power-state-change: SUCCESS: config-power-state-change ran successfully
2018-07-11 13:13:09,500 - main.py[DEBUG]: Ran 10 modules with 0 failures
2018-07-11 13:13:09,500 - util.py[DEBUG]: Creating symbolic link from '/run/cloud-init/result.json' => '../../var/lib/cloud/data/result.json'
2018-07-11 13:13:09,500 - util.py[DEBUG]: Reading from /proc/uptime (quiet=False)
2018-07-11 13:13:09,501 - util.py[DEBUG]: Read 16 bytes from /proc/uptime
2018-07-11 13:13:09,501 - util.py[DEBUG]: cloud-init mode 'modules' took 1144.781 seconds (1144.78)
2018-07-11 13:13:09,501 - handlers.py[DEBUG]: finish: modules-final: SUCCESS: running modules for final

/var/log/cfn-init.log:

2018-07-11 12:56:14,277 [INFO] -----------------------Build complete-----------------------
2018-07-11 13:13:09,306 [DEBUG] CloudFormation client initialized with endpoint https://cloudformation.us-east-1.amazonaws.com
2018-07-11 13:13:09,307 [DEBUG] Signaling resource AnsibleConfigServer in stack xyz-openshift-OpenShiftStack-1588DK7OO64YH with unique ID i-0bba4695fbe893afe and status FAILURE
```

Any ideas what could be going wrong?

**Comment by jaymccon** [comment]
There is an upstream redhat issue causing this: https://github.com/openshift/openshift-ansible/issues/7596 a fix has been merged, but we're waiting on a release to be published by the openshift ansible so that we can update.

**Comment by bczoma** [comment]
@jaymccon , do you have an ETA?

**Comment by jsm174** [comment]
@jaymccon Thanks for the info.

Prior to your comment, we tried hosting our own version of the quickstart in s3, with an overridden  `CURRENT_PLAYBOOK_VERSION` in `ansibleconfigserver_bootstrap.sh` since https://github.com/openshift/openshift-ansible/issues/7596 says it was fixed with https://github.com/openshift/openshift-ansible/pull/9125:

```
CURRENT_PLAYBOOK_VERSION=https://github.com/openshift/openshift-ansible/archive/openshift-ansible-${OCP_ANSIBLE_RELEASE}.tar.gz
curl  --retry 5  -Ls ${CURRENT_PLAYBOOK_VERSION} -o openshift-ansible.tar.gz
tar -zxf openshift-ansible.tar.gz
```

to

```
CURRENT_PLAYBOOK_VERSION=https://github.com/openshift/openshift-ansible/archive/release-3.9.tar.gz
curl  --retry 5  -Ls ${CURRENT_PLAYBOOK_VERSION} -o openshift-ansible.tar.gz
tar -zxf openshift-ansible.tar.gz
```



We launched the stack with:

```
aws cloudformation create-stack \
   --disable-rollback \
   --capabilities=CAPABILITY_IAM \
   --stack-name ${OS_PROJECT}-openshift \
   --template-url https://temp-awsqs.s3.amazonaws.com/quickstart-redhat-openshift/templates/openshift-master.template \
   --parameters "ParameterKey=AvailabilityZones,ParameterValue='${AWS_OS_AVAILABILITY_ZONES}'" \
                ParameterKey=ContainerAccessCIDR,ParameterValue="0.0.0.0/0" \
                ParameterKey=RemoteAccessCIDR,ParameterValue="0.0.0.0/0" \
                ParameterKey=KeyPairName,ParameterValue="${OS_PROJECT}-key" \
                ParameterKey=DomainName,ParameterValue="${AWS_ROUTE53_DOMAIN}" \
                ParameterKey=HostedZoneID,ParameterValue="${AWS_ROUTE53_HOSTED_ZONE_ID}" \
                ParameterKey=SubDomainPrefix,ParameterValue="${OS_PROJECT}" \
                ParameterKey=RedhatSubscriptionPoolID,ParameterValue="${REDHAT_POOL_ID}" \
                ParameterKey=RedhatSubscriptionUserName,ParameterValue="${REDHAT_USERNAME}" \
                ParameterKey=RedhatSubscriptionPassword,ParameterValue="${REDHAT_PASSWORD}" \
                ParameterKey=OpenShiftAdminPassword,ParameterValue="${OS_ADMIN_PASSWORD}" \
                ParameterKey=OpenshiftContainerPlatformVersion,ParameterValue="3.9" \
                ParameterKey=HawkularMetrics,ParameterValue="Disabled" \
                ParameterKey=QSS3BucketName,ParameterValue="temp-awsqs" \
                ParameterKey=QSS3KeyPrefix,ParameterValue="quickstart-redhat-openshift/"
```

It seems like it is picking up the latest 3.9 version of the playbook, but ultimately it is still failing in the same place:

```
/var/log/messages:

Jul 12 13:02:37 ip-10-0-134-248 cloud-init: + CURRENT_PLAYBOOK_VERSION=https://github.com/openshift/openshift-ansible/archive/release-3.9.tar.gz
Jul 12 13:02:37 ip-10-0-134-248 cloud-init: + curl --retry 5 -Ls https://github.com/openshift/openshift-ansible/archive/release-3.9.tar.gz -o openshift-ansible.tar.gz
Jul 12 13:02:38 ip-10-0-134-248 cloud-init: + tar -zxf openshift-ansible.tar.gz
Jul 12 13:02:38 ip-10-0-134-248 cloud-init: + mkdir -p /usr/share/ansible
Jul 12 13:02:38 ip-10-0-134-248 cloud-init: + mv openshift-ansible-release-3.9 /usr/share/ansible/openshift-ansible
.
.
.
Jul 12 13:07:08 ip-10-0-134-248 cloud-init: + '[' 3.9 == 3.9 ']'
Jul 12 13:07:08 ip-10-0-134-248 cloud-init: + ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/prerequisites.yml
Jul 12 13:07:10 ip-10-0-134-248 cloud-init: [WARNING]: Could not match supplied host pattern, ignoring: oo_lb_to_config
Jul 12 13:07:10 ip-10-0-134-248 cloud-init: [WARNING]: Could not match supplied host pattern, ignoring: oo_nfs_to_config
Jul 12 13:08:40 ip-10-0-134-248 cloud-init: [WARNING]: Could not match supplied host pattern, ignoring:
Jul 12 13:08:40 ip-10-0-134-248 cloud-init: oo_hosts_containerized_managed_true
Jul 12 13:10:24 ip-10-0-134-248 cloud-init: + ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/deploy_cluster.yml
Jul 12 13:10:28 ip-10-0-134-248 cloud-init: [WARNING]: Could not match supplied host pattern, ignoring: oo_lb_to_config
Jul 12 13:10:28 ip-10-0-134-248 cloud-init: [WARNING]: Could not match supplied host pattern, ignoring: oo_nfs_to_config
Jul 12 13:14:05 ip-10-0-134-248 cloud-init: ERROR! |failed expects hostvars is a dict
Jul 12 13:14:05 ip-10-0-134-248 cloud-init: + qs_err ' Openshift installation failed, see logs for details '
Jul 12 13:14:05 ip-10-0-134-248 cloud-init: + touch /var/tmp/stack_failed
Jul 12 13:14:05 ip-10-0-134-248 cloud-init: + echo '[FAILED] @  Openshift installation failed, see logs for details '
Jul 12 13:14:05 ip-10-0-134-248 cloud-init: + echo '[FAILED] @  Openshift installation failed, see logs for details '
Jul 12 13:14:05 ip-10-0-134-248 cloud-init: [FAILED] @  Openshift installation failed, see logs for details
Jul 12 13:14:05 ip-10-0-134-248 cloud-init: ++ qs_status
```

**Comment by jsm174** [comment]
And upon further review, even though we did pull down the latest 3.9 playbook, somehow the old version is still there:

```
diff /usr/share/ansible/openshift-ansible/roles/lib_utils/filter_plugins/openshift_master.py \
   /usr/share/ansible/openshift-ansible/openshift-ansible-release-3.9/roles/lib_utils/filter_plugins/openshift_master.py

5a6
> import collections
486c487,488
<         if not issubclass(type(hostvars), dict):
---
>         if not issubclass(type(hostvars), dict) \
>                 and not issubclass(type(hostvars), collections.Mapping):

**Comment by jaymccon** [comment]
@jsm174 thanks for your efforts on this one, I'm working on a fix that incorporates your findings, as well as locks to ansible-2.4.6 for openshift 3.7 support.

**Comment by danni-m** [comment]
@jaymccon 
I would like to re-raise the issue discussed here regarding permissions to the ansible instance.
It seems that `SetupRole` doesn't have access to cloudformation and cannot determine if the ASG setup is finished or not (see attached 
![image](https://user-images.githubusercontent.com/7194784/43151905-ad8d0dbe-8f75-11e8-9194-fbe96eca47ea.png)
I was able to setup the cluster by adding `AWSCloudFormationReadOnlyAccess` manually to `SetupRole`.

I created a PR with the appropriate fix to the template.

**Comment by blakemorgan** [comment]
I am having the same issue. It occurs regardless of if I try to use OpenShift 3.10 and 3.11. 

`The following resource(s) failed to create: [OpenShiftNodeASG, AnsibleConfigServer, OpenShiftEtcdASG, OpenShiftMasterASG]`

I am new to CloudFormation and don't know how to go about debugging or fixing this.

**Comment by kanikasingh-1490** [comment]
@blakemorgan were you able to resolve this issue?
Iam facing the same issue while creating stack using quickstart template

**Comment by jaymccon** [comment]
@danni-m @blakemorgan @kanikasingh-1490 I'm sorry to hear that you're having issues. The reason this particular issue was closed is that it is too generic to be affective. Can I please encourage you to have a look for more specific issues to add your "me too". If there is nothing directly relevant, please open a new issue.

In order to establish a more specific cause of a stack failure due to an EC2 instance or Autoscaling group failing with the error `Received FAILURE signal with UniqueId ...` you will need to look at the instance logs, which are available in CloudWatch Logs even if the stack has rolled back. Here are some AWS cli commands to simplify getting at the relevant logs. Be sure to replace the values in the 1st two lines with your CloudFormation stack name and AWS Region where the stack was launched.

```bash
STACK_NAME=<MyStackName>
REGION=<us-east-1>

INNER_STACK=$(aws cloudformation describe-stack-resources \
    --stack-name $STACK_NAME \
    --region $REGION \
    --query 'StackResources[?LogicalResourceId==`OpenShiftStack`].PhysicalResourceId' \
    --output text)
LOG_GROUP=$(aws cloudformation describe-stack-resources \
    --stack-name $INNER_STACK \
    --region $REGION \
    --query 'StackResources[?LogicalResourceId==`OpenshiftLogGroup`].PhysicalResourceId' \
    --output text)
for i in $(aws cloudformation describe-stack-events \
    --stack-name $INNER_STACK \
    --region $REGION \
    --query 'StackEvents[?ResourceStatusReason!=`null`]|[?starts_with(ResourceStatusReason, `Received FAILURE signal with UniqueId`) == `true`].PhysicalResourceId' \
    --output text) 
do 
    aws logs filter-log-events \
        --output text \
        --log-group-name $LOG_GROUP \
        --region $REGION \
        --start-time 0 \
        --filter-pattern '"cloud-init"' \
        --query 'events[].[logStreamName, message]' \
        --log-stream-names ${i}/var/log/messages
done
```

If you would like help, please be sure to include these logs  as well as:

* which launch option (New or Existing VPC)
* Date launched
* parameters used to launch
* CloudFormation events for all stacks (New VPC stack has 2 child stacks)

**Be sure to redact any sensitive information like RedHat logins, passwords, etc from logs, parameters and events**
